autoload/jedi.vim

```diff
@@ -392,20 +461,61 @@ endfunction
 " helper functions
 " ------------------------------------------------------------------------
 
-function! jedi#add_goto_window(len) abort
-    set lazyredraw
-    cclose
+" a:mode: 'goto' or 'usages'
+function! jedi#add_goto_window(mode, len, ...) abort
+    let select_entry = a:0 ? a:1 : 0
     let height = min([a:len, g:jedi#quickfix_window_height])
-    execute 'belowright copen '.height
-    set nolazyredraw
-    if g:jedi#use_tabs_not_buffers == 1
-        noremap <buffer> <CR> :call jedi#goto_window_on_enter()<CR>
-    endif
-    augroup jedi_goto_window
-      au!
-      au WinLeave <buffer> q  " automatically leave, if an option is chosen
-    augroup END
-    redraw!
+
+    " Using :cwindow allows to stay in the current window in case it is opened
+    " already.
+    execute 'belowright cwindow '.height
+
+    if &filetype ==# 'qf'
+        " Setup quickfix window, but only if it was opened / initially.
```

Comment:
Can you comment a bit more on when it is opened and when not? I didn't really understand that. Also this if probably means that the focus is now in the quickfix window is selected (which it is only in goto mode?). I'm just not really sure when this triggers.

Reply:
Changed the comment to:

> Setup quickfix window (initially, when opened via ":cwindow" above).

Clear?

Reply:
I guess I don't understand when this is not the case, because cwindow gets executed. While I know that cwindow can also close the window, I don't understand when that would trigger.

Reply:
The window might be open already.
Then cwindow does nothing.
Happens when using usages again while it's open, and if there's a window already (I guess).
I.e. that checks if we've moved to the window really.